### PR TITLE
Inbox notification: Remove icons

### PR DESCRIPTION
**Pull request by octaedro**
Fixes #partially-4099

In this PR the `icon` is removed from all the inbox notifications.

_Please be sure the [PR 4256](https://github.com/woocommerce/woocommerce-admin/pull/4256) has been approved before approving this one_

> **_NOTE:_**  As this is a part of a big functionality this branch will be merged with the branch `add/4099`, and it will be that branch that will be merged with `master`.
To make the code reviewing easier, this PR only has the code referring to the functionality described in the title.
If you want to test everything together, all the changes are merged in the branch: `add/4099_test_branch`.

### Screenshots
screenshot-one wordpress test-2020 04 29-17_52_02

### Detailed test instructions:

- dbDelta should remove the `icon` column from the table `wp_wc_admin_notes`.
- Verify that the inbox notifications `icons` are not present anymore in the frontend.
- The same with the backend.

### Changelog Note:

Dev: Removed from all the inbox notifications

### CODE REVIEW

File: client/header/activity-panel/panels/inbox/card.js [hunk 1]
```diff
@@ -3,7 +3,6 @@
  */
 import { Component } from '@wordpress/element';
 import PropTypes from 'prop-types';
-import Gridicon from 'gridicons';
```
Comment (joshuatf) [inline client/header/activity-panel/panels/inbox/card.js:6]: I see `<GridIcon />` still being used in this file and some icon related settings here which prevents my page from loading.

Following this down, I notice that icon is still being used in `ActivityCard`.  Just making sure that this is only meant to affect inbox notes and not other activity cards, is that right?

Comment (joshuatf) [reply client/header/activity-panel/panels/inbox/card.js:6]: Another note on this- the placeholder for `ActivityCard` has an "icon" div.  Not sure if we plan on leaving this or not for other panels using the activity card.

Ideally, we could pass a boolean `icon` prop which would determine if the placeholder shows this or not.

Comment (octaedro) [reply client/header/activity-panel/panels/inbox/card.js:6]: Hi @joshuatf,

I think that after merging the [PR 4256](https://github.com/woocommerce/woocommerce-admin/pull/4256) with `add/4099` we will have the missing elements that are blocking the normal behavior of this branch. For this purpose, I created the test_branch (`add/4099_test_branch`) that has all the changes together.

>Following this down, I notice that icon is still being used in ActivityCard. Just making sure that this is only meant to affect inbox notes and not other activity cards, is that right?

Yes, exactly.
I tried to decouple the `InboxNoteCard` from the `ActivityCard`. We still continue using the `ActivityCard` with orders and reviews but not with inbox notifications anymore.
My plan was trying to have an `InboxNoteCard` that we could use as an ActivityCard in the InboxPanel or eventually in the dashboard with no big modifications.

`Gridicon` is not used anymore In the `card.js`.

>Another note on this- the placeholder for ActivityCard has an "icon" div. Not sure if we plan on leaving this or not for other panels using the activity card.

This is a good point. I left the `ActivityCardPlaceholder` -when is loading- and an empty `ActivityCard` -when we don't have results- in the inbox panel, to follow the expected behavior for the inbox panel. In the Home screen it's different. If we don't have results the whole inbox messages column will disappear. For the loading situation, I wasn't sure if we will use something specific or the same `ActivityCardPlaceholder`.

I hope have responded to all the questions you had.

If you think we should change the approach just let me know and I will change it 🙂

**Comment by jeffstieler** [comment]
I'm still seeing the icon in the loading placeholders.

Screen Shot 2020-05-12 at 5 37 27 PM

**Comment by octaedro** [comment]
Hi @jeffstieler,

Yes, in this PR I only removed the icon from the inbox message. 

The inbox panel in this PR shows 3 kinds of cards:
1. The `InboxNoteCard`.
2. A generic `ActivityCard` (when there aren't cards to show -it uses an icon-).
3. And the `ActivityCardPlaceholder` (when it's loading the messages).

After having the new designs for the empty inbox notification and the placeholders, I created the [PR 4379](https://github.com/woocommerce/woocommerce-admin/pull/4379).
That PR removes the icons for every situation in the inbox panel.

Thank you for the alert about this problem Jeff 🙂

**Pull request merged**